conans/client/generators/cmaketoolchain.py

```diff
@@ -0,0 +1,88 @@
+import shlex
+import os
+
+from conans.paths import BUILD_INFO_CMAKETOOLCHAIN
+from conans.model import Generator
+
+class CMakeToolchainGenerator(Generator):
+    @property
+    def filename(self):
+        return BUILD_INFO_CMAKETOOLCHAIN
+
+    def cmake_system_name(self):
+        names = {
+            "Macos": "Darwin",
+            "iOS": "Darwin",
+        }
+
+        osname = str(self.conanfile.settings.os)
+
+        if osname in names:
+            return names[osname]
+
+        return osname
+
+    @property
+    def content(self):
+
+        def make_absolute(prg):
+            if os.path.isabs(prg):
+                return prg
+
+            for path in os.environ.get("PATH", "").split(os.pathsep):
+                binpath = os.path.join(path, prg)
+                if os.path.exists(binpath):
+                    return str(binpath)
+
+            return prg
+
+        def split_ccache(env, default=""):
```

Comment:
Do you have some example of what is exactly doing this?

Reply:
It takes a string like `ccache gcc` and splits it into `CMAKE_C_COMPILER=ccache` and `CMAKE_C_COMPILER_ARG1=gcc`. CMake internally always assumes that CMAKE_C_COMPILER is a path to a binary - not a command with multiple arguments.

Reply:
I am not sure on other tools besides ccahe wrapping compiler, but may be it can be generalized to handle N arguments?

Reply:
Yeah well. This is actually handled this way. The cmake variable is strangely named - arg1 handles like a normal shell invocation